netcore/System.Private.CoreLib/src/System/RuntimeType.Mono.cs

```diff
@@ -1875,19 +1876,24 @@ internal RuntimeType (Object obj)
 		internal RuntimeConstructorInfo GetDefaultConstructor ()
 		{
 			var cache = Cache;
-			RuntimeConstructorInfo ctor = cache.default_ctor;
+			RuntimeConstructorInfo ctor = null;
 
-			if (ctor == null) {
-				var ctors = GetConstructors (BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.DeclaredOnly);
+			if (cache.default_ctor_cached) {
+				return cache.default_ctor;
```

Comment:
No need for {}.

Reply:
Note that having two variables instead of one makes this code no longer thread safe without barriers etc.